Neaten Padname flag contants

leonerd opened the pull request:
The constants were named `PADNAMEt_*` suggesting their purpose is a type enumeration. They're not. They're bitflags, so the name `PADNAMEf_*` would be more appropriate.

Also updated the `B` and `B::Deparse` modules to make use of the new modern names (and avoid the old `SVpad_*` flags).

ilmari commented:
There appears to at least one [CPAN module](https://grep.cpanauthors.org/search?q=PADNAMEt_&extension=xs) using the `PADNAMEt_*` constants.

leonerd commented:
@ilmari Hah - typical that it would happen to be one of mine :)
OK I suppose I'll do those back-compat wrappers too.

ilmari commented on pad.h:
```diff
@@ -348,6 +348,11 @@ Restore the old pad saved into the local variable C<opad> by C<PAD_SAVE_LOCAL()>
 #define SVpad_STATE		PADNAMEf_STATE
 #define SVpad_TYPED		PADNAMEf_TYPED
 #define SVpad_OUR		PADNAMEf_OUR
+#define PADNAMEt_OUTER          PADNAMEf_OUTER
+#define PADNAMEt_STATE          PADNAMEf_STATE
+#define PADNAMEt_LVALUE         PADNAMEf_LVALUE
+#define PADNAMEt_TYPED          PADNAMEf_TYPED
+#define PADNAMEt_OUR            PADNAMEf_OUR
```
To avoid accidental use in-core, these should be wrapped in `#ifndef PERL_CORE`.

leonerd replied:
Huh. Wow it turns out quite a few bits of existing code in core were still even using those old names from the previous round of renaming. I'll fix those up too.

ilmari replied:
If I were being _really_ nitpicky I would do the commits in the opposite order, but I shan't ;-)

leonerd committed:
Guard the older `SvPAD*` wrappers with `#ifndef PERL_CORE`

leonerd committed:
Neaten the PADNAME flag constants

Rename the `PADNAMEt_*` constants to `PADNAMEf_*`, as they're supposed
to represent bitflags, not a type enumeration.

Also updated the `B` and `B::Deparse` modules to make use of the new modern
names (and avoid the old `SVpad_*` flags).

Also added `PADNAMEt_*` back-compat defines, guarded by `#ifndef PERL_CORE`
so as not to permit their use accidentally within perl core.
